bundle/config/mutator/process_target_mode.go

```diff
@@ -148,6 +157,14 @@ func isRunAsSet(r config.Resources) bool {
 	return true
 }
 
+func isExplicitRootSet(b *bundle.Bundle) bool {
```

Comment:
It can also be explicit if configured in the root configuration.

I would prefer a solution that doesn't rely on keeping the targets around to figure this out. It could set a flag on the bundle struct (not the config but `*bundle.Bundle`) to capture if the field was set explicitly instead of keeping the targets around to figure it out. Then it can be configured either in the root configuration or in the target with the same effect.

Reply:
> It can also be explicit if configured in the root configuration.

I'd really prefer to just have users set the `workspace_path` in the target though, that makes it much harder to shoot yourself in the foot. If it's in the root then other targets may be using the same `workspace_root`. The strictness of `mode: production` seems like a fine place to make sure you just set a `workspace_root` in the target?